packages/cli/src/commands/integration/list.ts

```diff
@@ -0,0 +1,118 @@
+import chalk from 'chalk';
+import type Client from '../../util/client';
+import getScope from '../../util/get-scope';
+import { getLinkedProject } from '../../util/projects/link';
+import type { Store } from './types';
+import { getResources } from '../../util/integration/get-resources';
+import { listSubcommand } from './command';
+import { getFlagsSpecification } from '../../util/get-flags-specification';
+import { parseArguments } from '../../util/get-args';
+import handleError from '../../util/handle-error';
+import table from '../../util/output/table';
+
+export async function list(client: Client) {
+  let parsedArguments = null;
+  const flagsSpecification = getFlagsSpecification(listSubcommand.options);
+
+  try {
+    parsedArguments = parseArguments(client.argv.slice(3), flagsSpecification);
+  } catch (error) {
+    handleError(error);
+    return 1;
+  }
+
+  const { contextName, team } = await getScope(client);
+
+  if (!team) {
+    client.output.error('Team not found');
+    return 1;
+  }
+
+  const project = await getLinkedProject(client).then(result => {
+    if (result.status === 'linked') {
+      return result.project;
+    }
+
+    return null;
+  });
+
+  if (parsedArguments.flags['--current-project'] && !project) {
```

Comment:
Do we apply a similar pattern with `--current-project` in other commands?

Reply:
So, this is a bit of an interesting one. Often commands are either untethered to a linked project at all (ex. redeploy takes a deployment URL you specify as an argument), or are in the pure context of the linked project. integration is a bit interesting because it is both team-wide and linked to projects.

`vc list` seems like it might be the closest, in that `vc list` takes argument `[project-id]` afterwards to specify which project you want to list the details of, and if you leave it blank, it uses the current linked project.

But we also have the scope above which is the team-wide list of resources, and our original outline for what we want `vc integration list` to do is to view the team-wide list. The `--current-project` flag was just my small addition to make it a little more usable.

Ignoring the option of "this is fine as-is," the immediate alternatives I can think of that would fit are:
1. Have no team scope, only ever project scope (this loses the opportunity to search across an integration and makes some debugging harder from CLI).
2. Have current project scope by default, an optional argument to specify a project ID, and add an `--all`/`--team` flag to expand to team scope (which fails if there's a project ID argument given). Further possible expansion: if there is no current project, default behaviour of no argument becomes team scope?

I'm not a fan of option 1, but I'm all right with option 2 for being more similar to unix `ls [-a]` functionality.

Reply:
Yes, the inverse is what i was wondering about (2). I'm ok either way, but wanted to make sure this was discussed with CLI owners.